src/Nethermind/Nethermind.Core/Crypto/PublicKey.cs

```diff
@@ -96,7 +97,17 @@ public override bool Equals(object? obj)
 
         public override int GetHashCode()
         {
-            return MemoryMarshal.Read<int>(Bytes);
+            byte[] bytes = Bytes;
+            long l0 = Unsafe.ReadUnaligned<long>(ref MemoryMarshal.GetArrayDataReference(bytes));
+            long l1 = Unsafe.ReadUnaligned<long>(ref Unsafe.Add(ref MemoryMarshal.GetArrayDataReference(bytes), sizeof(long)));
```

Comment:
wouldn't it be faster if we had `offset` and then add `sizeof(long)` on each step?

Reply:
Not really as that would introduce a variable; here they are all consts